templates/master/00-master/baremetal/files/baremetal-haproxy.yaml

```diff
@@ -109,12 +109,12 @@ contents:
           privileged: true
         image: {{ .Images.baremetalRuntimeCfgImage }}
         command:
-        - monitor
-        - "/var/lib/kubelet/kubeconfig"
-        - "/config/haproxy.cfg.tmpl"
-        - "/etc/haproxy/haproxy.cfg"
-        - "--api-vip"
-        - "{{ .Infra.Status.PlatformStatus.BareMetal.APIServerInternalIP }}"
+          - "/bin/bash"
+          - "-c"
+          - |
+            #/bin/bash
+            cp /host/etc/resolv.conf /etc/resolv.conf
```

Comment:
what about this non-atomic copy? that's another race

Reply:
@rgolangh  I don't think that atomic cp is required here,  the monitor code runs periodically so even if it fails once or twice  (till the CP operation completed) it should be OK.

Reply:
I'm not worried about the monitor, but on the dns resolver that could have a resolv.conf bits in-flight. I'm quite sure this is what happened on the old prepender.

Reply:
Do you mean the resolv.conf in the host or in the pod?

Reply:
better to do it atomically

Reply:
@celebdor @rgolangh 
I tried to change cp to be atomic ( use the same trick as [1] ) , but mv command fails with EBUSY error because /etc/rsolv.conf is a mount point.

Do you have any idea how to make cp atomic?

[1] https://github.com/openshift/machine-config-operator/pull/1763/files

Reply:
I looked into this a bit more. First, I don't think we can make operations on resolv.conf atomic because it is its own tmpfs filesystem inside the container. Even mv is not atomic across filesystems (it degrades into a cp+unlink).

However, I don't think this is a concern inside the container. As Yossi pointed out, if we get an incomplete resolv.conf from the copy then the liveness probe will immediately fail because of the mismatch with the host resolv.conf. Furthermore, nothing is running in the container at the point where the copy is made so there's no possibility of in-flight DNS resolutions using a bad resolv.conf. The monitor doesn't start until after the file has been copied. The important thing here is that we're not modifying the system-wide resolv.conf, just this one container's, so there's no potential problems with concurrent access to the file.